### PR TITLE
Safer payments

### PR DESCRIPTION
**Pull request by zorkian**
Fix an issue where preload_props could get no database handle and
wouldn't fail immediately (but would instead fail two lines later).
Also, make the payment system catch failures in the post-payment tasks
and not fail hard.

### CODE REVIEW

File: cgi-bin/LJ/User/Account.pm [hunk 1]
```diff
@@ -822,6 +822,8 @@ sub preload_props {
                 LJ::get_db_reader();
             $used_slave = 1;
         }
+        die "No db\n" unless $db;
+
```
Comment (kareila) [inline cgi-bin/LJ/User/Account.pm]: This should probably either give a bit more context, or remove the "\n" so that the line number gets printed.

**Comment by zorkian** [comment]
Updated to use `confess` which prints the backtrace, infinitely more useful

**Comment by zorkian** [comment]
Going to assume this is OK and merge

**Comment by kareila** [comment]
Yeah, I haven't tested it but it looked fine to me. Thanks for tracking it down.

**Comment by rahaeli** [comment]
shit, I wonder if this might be the cause of the occasional mysterious failures that we've been trying to figure out for like FOUR GODDAMN YEARS? A GIRL CAN HOPE

**Pull request merged**